source/common/http/conn_manager_impl.cc

```diff
@@ -1050,6 +1051,7 @@ void ConnectionManagerImpl::ActiveStream::encode100ContinueHeaders(
       return;
     }
   }
+  current_encoder_filter_ = nullptr;
```

Comment:
I think we have to care about the early return on 1051 etc.  May want some wrapper shim that goes on the stack and nulls when it goes out of scope.  Though again I think that'll have the same problems with reentrant code.

Reply:
Good catch! I fixed it by resetting current_encoder_filter_ right after calling filters' handling function. This way, we may not need to worry about early return. current_encoder_filter_ is not null only when we call filters' handling function, and is set to null immediately after.